Sources/CodeGeneratorExecutable/CodeGenerator.swift

```diff
@@ -0,0 +1,99 @@
+import Foundation
+
+@main
+struct CodeGenerator {
+    static func main() throws {
+        // Use swift-argument-parser or just CommandLine, here we just imply that 2 paths are passed in: input and output
+        guard CommandLine.arguments.count == 2 else {
+            throw CodeGeneratorError.invalidArguments
+        }
+        // arguments[0] is the path to this command line tool
+        let output = URL(filePath: CommandLine.arguments[1])
+
+        // generate default implementations of RealFunctions for SIMD protocol
+        let realFunctionSIMDFileURL = output.appending(component: "SIMD+RealFunctions.swift")
+        let realFunctionsSIMDExtension = RealFunctionsGenerator.realFunctionsExtension(
+            objectType: "SIMD",
+            type: "Self",
+            whereClause: true,
+            simdAccelerated: false
+        )
+        try realFunctionsSIMDExtension.write(to: realFunctionSIMDFileURL, atomically: true, encoding: .utf8)
+
+        let floatingPointTypes: [String] = ["Float", "Double"]
+        let simdSizes: [Int] = [2, 4, 8, 16, 32, 64]
+
+        for floatingPointType in floatingPointTypes {
+            // Generator Derivatives for RealFunctions for floating point types
+            let realFunctionDerivativesFileURL = output.appending(
+                component: "\(floatingPointType)+RealFunctions+Derivatives.swift",
+                directoryHint: .notDirectory
+            )
+            let type = floatingPointType
```

Comment:
This is ... odd? Is this changing the type from String to something else (in which case annotate it?); or just defining a shorthand name; or ...?

Reply:
ah that was there to match the same structure as around line 65. Removed it now.